lib/ABI/FunctionType/Conversion.cpp

```diff
@@ -438,32 +445,102 @@ TCC::tryConvertingStackArguments(model::QualifiedType StackArgumentTypes,
                                       CurrentArgument.Offset(),
                                       TheNextOne.Offset(),
                                       NextAlignment)) {
-      return std::nullopt;
+      // We met a argument we cannot just "add".
+      break;
     }
 
     // Create the argument from this field.
     model::Argument &New = Result.emplace_back();
     model::copyMetadata(New, CurrentArgument);
     New.Index() = DistributionHelper.ArgumentIndex - 1;
     New.Type() = CurrentArgument.Type();
+
+    CurrentRange = std::ranges::subrange(std::next(CurrentRange.begin()),
+                                         Stack.Fields().end());
+  }
+
+  // The main loop is over. Which means that we either converted everything but
+  // the very last argument correctly, OR that we aborted half-way through.
+  if (CurrentRange.size() == 1) {
+    // Having only one element in the "remaining" range means that only the last
+    // field is left - add it too after checking.
+    const model::StructField &LastArgument = CurrentRange.front();
+
+    // This is a workaround for the cases where expected alignment is missing
+    // at the very end of the RFT-style stack argument struct:
+    uint64_t FullSize = abi::FunctionType::paddedSizeOnStack(Stack.Size(),
+                                                             FullAlignment);
+
+    std::optional<uint64_t> LastSize = LastArgument.Type().size();
+    revng_assert(LastSize.has_value() && LastSize.value() != 0);
+    if (DistributionHelper.canBeNext(LastArgument.Type(),
+                                     LastArgument.Offset(),
+                                     FullSize,
+                                     FullAlignment)) {
+      model::Argument &New = Result.emplace_back();
+      model::copyMetadata(New, LastArgument);
+      New.Type() = LastArgument.Type();
+      New.Index() = DistributionHelper.ArgumentIndex - 1;
+      return Result;
+    }
+  }
+
+  // Getting to this point (past the return statement in the last element
+  // section) means that there is at least one argument we cannot just "add".
+  revng_log(Log,
+            "Naive conversion failed. Try to fall back on using structs "
+            "instead.");
```

Comment:
> `"ToCFT: use structs as a fallback"`: report one or more specific examples of things that we did not support and that we now support thanks to such commit. Put the example in the commit message too.

cc: @aleclearmind

Any function with stack arguments that skipped registers was broken before this. Even something as simple as
```cpp
struct three_words { uint64_t a, b, c; };
void function(struct three_words s);
```

We didn't have to _handle_ all of them right away, but detection is >90% of the work. Once we have all the information necessary to know that what we were about to emit is wrong, we might as well emit the right thing.